hashing/src/indexed_merkle_proof.rs

```diff
@@ -393,6 +398,17 @@ mod test {
         Digest::hash_pair(count.to_le_bytes(), raw_root)
     }
 
+    // Returns random digest
+    fn random_digest() -> Digest {
+        let mut rng = rand::thread_rng();
+        let total_bytes = rng.gen::<u8>() as usize;
+        Digest::hash(
+            iter::repeat_with(|| rng.gen::<u8>())
+                .take(total_bytes)
+                .collect_vec(),
+        )
+    }
```

Comment:
Maybe it would make sense to make this public, not test-only, and more general, so we'll be able to use it e.g. in the `node` crate with the `TestRng` as well. ([It's there on `feat-fast-sync`](https://github.com/casper-network/casper-node/blob/eff7252aeca274505ecd315041ed2453d424bafb/node/src/crypto/hash.rs#L79), so it looks like we'll need it.)

```suggestion
    pub fn random<R: Rng>(rng: &mut R) -> Self {
        Digest(rng.gen::<[u8; Digest::LENGTH]>())
    }
```

Reply:
Changed to use the `rand::Distribution` trait.